classes/class-gf-securesubmit.php

```diff
@@ -1,950 +1,1597 @@
+                $field['failed_validation']  = true;
+                $field['validation_message'] = $this->getSecureSubmitJsError();
+            }
+            else {
+                $field['failed_validation'] = false;
+            }
+
+            break;
+        }
+
+        $validationResult['is_valid'] = true;
+
+        return parent::maybe_validate($validationResult);
+    }
+
+    /**
+     * @param $validation_result
+     *
+     * @return mixed
+     */
+    public
+    function validation($validation_result)
+    {
+        if (!rgar($validation_result['form'],'id',false)) {
+            return $validation_result;
+        }
+        if (!$this->has_feed($validation_result['form']['id'], true)) {
+            return $validation_result;
+        }
+
+        $isACH = false;
+        $isCC = false;
+        foreach ($validation_result['form']['fields'] as $field) {
+            $current_page         = GFFormDisplay::get_source_page($validation_result['form']['id']);
+            $field_on_curent_page = $current_page > 0 && $field['pageNumber'] == $current_page;
+
+            if (GFFormsModel::get_input_type($field) == 'hpsACH' && $field_on_curent_page) {
+                $isACH = $field;
+                if (!$this->hasPayment($validation_result)) {
+                    $field['failed_validation']  = true;
+                    $field['validation_message'] = 'Please Check your entries and try again';
+                }
+                else {
+                    // override validation in case user has marked field as required allowing securesubmit to handle cc validation
+                    $field['failed_validation'] = false;
+                }
+            }
+
+            if (GFFormsModel::get_input_type($field) == 'creditcard' && $field_on_curent_page) {
+                $isCC = $field;
+                if ($this->getSecureSubmitJsError() && $this->hasPayment($validation_result)) {
+                    $field['failed_validation']  = true;
+                    $field['validation_message'] = $this->getSecureSubmitJsError();
+                }
+                else {
+                    // override validation in case user has marked field as required allowing securesubmit to handle cc validation
+                    $field['failed_validation'] = false;
+                }
+            }
+        }
+
+
+        $form  = $validation_result['form'];
+        $entry = GFFormsModel::create_lead($form);
+        $feed  = $this->get_payment_feed($entry, $form);
+
+        if (!$feed) {
+            return $validation_result;
+        }
+        $this->includeSecureSubmitSDK();
+
+        $submission_data = $this->get_submission_data($feed, $form, $entry);
+        $submission_data = array_merge($submission_data, $this->get_submission_dataACH($feed, $form, $entry));
+
+        if($isCC && !empty($submission_data['card_number']) && $isACH && !empty($submis
```

Comment:
@ttnnkkrr Please reference the [original implementation of this class](https://github.com/hps/heartland-gravity-forms-addon/blob/master/classes/class-gf-securesubmit.php). The override for `GFPaymentAddOn::validation` is meant for submission validation. The override for `GFPaymentAddOn::authorize` is meant for authorizing the payment method. It will be best if we keep these actions separate